### PR TITLE
feat: `equals()` should be nullable

### LINKED ISSUE #4: Check the operand of the `equals` method should be nullable
By convention, the operand of the `equals` method should be nullable. In other words, `a.equals(null)` should be valid.

### PR DESCRIPTION
**Pull request by Charm-Xin**
close https://github.com/ocpddev/jsensible/issues/4

### CODE REVIEW

File: src/main/kotlin/dev/ocpd/jsensible/internal/java/ParameterIsNullable.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.tngtech.archunit.lang.ArchCondition
+import com.tngtech.archunit.lang.ConditionEvents
+import com.tngtech.archunit.lang.SimpleConditionEvent.violated
+
+object ParameterIsNullable {
+
+    /**
+     * All parameters within the method should be annotated with @Nullable
+     */
+    fun parameterIsNullable(): ArchCondition<JavaMethod> =
```
Comment (unlimitedsola) [inline src/main/kotlin/dev/ocpd/jsensible/internal/java/ParameterIsNullable.kt:13]: Can we do this with `DescribedPredicate<JavaMethod>`?

**Comment by EggOxygen** [comment]
mark as draft since this is WIP

File: src/main/kotlin/dev/ocpd/jsensible/internal/java/ParameterIsNullable.kt (hunk 1 above)
Comment (Charm-Xin) [reply src/main/kotlin/dev/ocpd/jsensible/internal/java/ParameterIsNullable.kt:13]: I have tried, but it didn't go well.

**Comment by unlimitedsola** [comment]
Please resolve the conflicts and provide the test cases for this.

**Comment by unlimitedsola** [comment]
Getting no responses, I implemented this myself in fb478e7711d792c95df976be523573c2dea566b6

**Pull request closed without merging**